### PR TITLE
Add http connector and simplify the elastic search connector

### PR DESCRIPTION
**Pull request by liufan**
## Proposed changes

* added http connector
* removed the layer of IO operation in the ES connector and replaced with a batch iterator (IdAndVersionBatchResultIterator). hence removed fair amount of code in elasticsearch module
* moved the sorting out from the es module into the connector layer (AbstractIdAndVersionStreamConnector) and controlled by the connector config
* merged the IdAndVersionStream into Connector since there is no much logic left in it
* renamed module: `scrutineer`-> `cli`
* changed the group id and the package name to `com.aconex.scrutineer2`

## Checklist

- [x] I've read the [CONTRIBUTING](https://github.com/Aconex/scrutineer/blob/master/CONTRIBUTING.md) doc
- [x] All tests pass locally with my changes
- [x] I've added unit tests and integration tests.
- [x] I've added necessary documentation (if appropriate)

### CODE REVIEW

File: cli/config/example-jdbc.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# the connector class to use to create the stream
-stream.connector.class=com.aconex.scrutineer.jdbc.JdbcStreamConnector
+# the source type of the stream
+source.type=jdbc
```
[the remark this answers is not in the document]
Comment (tallpsmith) [reply cli/config/example-jdbc.properties:2]: There's some overlap with the work Rock was doing over in https://github.com/Aconex/scrutineer/pull/64/files# 

The Connector would be able to have properties resolved easily through different config providers.

I'd rather keep the `stream.connector.class` concept if we can, as long as we maintain that link to what the Connector interface is.  Is this something we can bring back?

**Comment by tallpsmith** [comment]
@liufan I think the checks are failing because this is based off the other branch that has the version number change done.  I haven't had a chance to merge that one, wondering if you would able to eyeball and give me your thoughts too ? (didn't want to do that by myself without another set of eyes).

see https://github.com/Aconex/scrutineer/tree/version_change

If you're happy with that branch, I'll merge it, and we can rebase this off master again?

**Review by tallpsmith — CHANGES_REQUESTED** [review]
Other than the removed/simplified connector enum, everything else looks ok.  I'm guessing you're trying to simplify it here to focus on our own needs, but removing some flexibility we're trying to allow in this library would be a shame.

**Comment by liufan** [comment]
> @liufan I think the checks are failing because this is based off the other branch that has the version number change done. I haven't had a chance to merge that one, wondering if you would able to eyeball and give me your thoughts too ? (didn't want to do that by myself without another set of eyes).
> 
> see https://github.com/Aconex/scrutineer/tree/version_change
> 
> If you're happy with that branch, I'll merge it, and we can rebase this off master again?

Thanks @tallpsmith . The branch looks good to me. BTW, in my commit, I also changed the group name to `com.aconex.scrutineer2`. I'll leave it to you to decide if we change it or not.

**Comment by tallpsmith** [comment]
> Thanks @tallpsmith . The branch looks good to me. BTW, in my commit, I also changed the group name to `com.aconex.scrutineer2`. I'll leave it to you to decide if we change it or not.

I had thought that it best to use the same `groupId` but change the `artifactId` would be sufficient.

**Comment by tallpsmith** [comment]
@liufan you probably want to rebase off master now that #65 is merged to master with the Version changes, and lets see where this is at with the CI pipeline.

File: cli/config/example-jdbc.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (liufan) [reply cli/config/example-jdbc.properties:2]: Sure, it's worth discussing. I realised I did not fully get the rationale of exposing the class name in the properties file when I made the change. Just thought the class name is internal implementation details, so I replaced it with a loose coupling enum.  
Just to confirm, the `stream.connector.class` is to provide the extension point for someone who want to create their own Connector without changing the Scrutineer code? The intension makes sense to me. Just wonder if there is better way to hide the class name of the build-in Connector for decoupling purpose. 
Supporting both `source.type` and `stream.connector.class` may be a bit overkill. I am Ok with either support both or just `stream.connector.class`.

[the remark this answers is not in the document]
Comment (tallpsmith) [reply cli/config/example-jdbc.properties:2]: Yes the `stream.connector.class` is the extension point, and each Connector config declares which one to use.

Longer term, we might look at a classpath scanning mechanism, similar to Spring, to look for other implementations, but it's just simpler at the moment to have the person creating the connector configuration file reference the right class name.  

If an integrator wants to use their own connector, they just need to ensure their code(JAR etc) is on the classpath, and reference their own classname in the connector configuration.

I think if we had to pick one, lets choose the Class for the moment.

**Pull request closed without merging**